src/analyses/goto_check.cpp

```diff
@@ -348,6 +349,39 @@ void goto_checkt::mod_by_zero_check(
     guard);
 }
 
+/// check a mod expression for the case INT_MIN % -1
+void goto_checkt::mod_overflow_check(const mod_exprt &expr, const guardt &guard)
+{
+  if(!enable_signed_overflow_check)
```

Comment:
`if(!enable_signed_overflow_check || mode == ID_java)` based on the conversation.

Reply:
ok